lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -818,10 +833,13 @@ private void checkIfTasksMatchTransferCriteria(List<String> taskIds, List<Minima
                 taskIdIterator.remove();
             }
         }
+        LOGGER.debug("exit from checkIfTasksMatchTransferCriteria()");
     }
 
     private void checkIfTasksMatchCompleteCriteria(List<String> taskIds, List<TaskSummary> taskSummaries,
         BulkOperationResults<String, TaskanaException> bulkLog) {
+        LOGGER.debug("entry to checkIfTasksMatchCompleteCriteria(taskIds = {}, taskSummaries = {}, bulkLog = {})",
+            LoggerUtils.listToString(taskIds), LoggerUtils.listToString(taskSummaries), bulkLog);
         Instant now = Instant.now();
```

Comment:
The LOGGER statement on line 841 should only be called if debug is enabled (to invoke listToString() only when required...